python/mlcroissant/mlcroissant/_src/operation_graph/execute.py

```diff
@@ -149,12 +154,69 @@ def execute_operations_in_beam(
         files = operation(files)
         if isinstance(operation, FilterFiles):
             break
-    pipeline = pipeline | "Shard by files" >> beam.Create(files)
+    if not isinstance(files, Iterable):
+        raise ValueError("Could not filter files.")
+    files = list(files)  # even for large datasets, this can be handled in RAM.
+
+    # We first shard by file and assign a shard_index.
+    pipeline = pipeline | "Shard by files with index" >> beam.Create(enumerate(files))
+    num_shards = len(files)
+
+    # We don't know in advance the number of records per shards. So we just allocate the
+    # maximum number which is `sys.maxsize // num_shards`. For a large evenly
+    # distributed dataset, we have:
```

Comment:
NIT: Maybe:

"Taking the practical case of large evenly distributed datasets on HuggingFace, we can compute the following:"

or something similar? Currently it is a bit confusing for a user reading about very specific HF numbers in the context of a generic function.

Reply:
Done.